thanos-operator.yaml

```diff
@@ -62,7 +53,33 @@ update:
     identifier: banzaicloud/thanos-operator
 
 test:
+  environment:
+    contents:
+      packages:
+        - kubectl
+        - git
   pipeline:
     # AUTOGENERATED
     - runs: |
         manager --help
+    - uses: test/kwok/cluster
+    - runs: |
+        git clone --depth=1 https://github.com/banzaicloud/thanos-operator.git
+        kubectl apply --validate=false -f thanos-operator/charts/thanos-operator/crds/
+    - name: Launch thanos-operator with dummy kubeconfig
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          export KUBERNETES_SERVICE_HOST="127.0.0.1"
+          export KUBERNETES_SERVICE_PORT="8080"
+
+          # Create required certificates directory
+          mkdir -p /tmp/certs
+          kubectl config view --minify --raw > /tmp/kwok-kubeconfig.yaml
+        start: |
+          manager -metrics-addr=:8080 -enable-leader-election=false -kubeconfig=/tmp/kwok-kubeconfig.yaml -verbosity=2
+        timeout: 30
+        expected_output: |
+          starting manager
+          starting metrics server
+          Starting Controller
```

Comment:
Can we have a `post:` step here, and check for port with `wait-for-it` and simple `curl` request, if possible?

Reply:
@Dentrax updated package PR to include a `post` step that checks the metrics port 👍